.github/workflows/terraform-azdev-agent-release.yml

```diff
@@ -0,0 +1,55 @@
+name: Release azure devops agent
+
+on:
+  push:
+    tags:
+      - "tf-azdev-agent-*"
+
+permissions:
+  contents: read
+  packages: write
+
+jobs:
+  release:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - name: Get variables from tags
+        id: vars
+        run: |
+          tag=${GITHUB_REF/refs\/tags\/tf-azdev-agent-/}
+          echo "version=${tag%-demo}" >> "$GITHUB_OUTPUT"
+          echo "reponame=${GITHUB_REPOSITORY,,}" >> "${GITHUB_OUTPUT}"
+      - name: Get Git commit timestamps
+        run: echo "TIMESTAMP=$(git log -1 --pretty=%ct)" >> $GITHUB_ENV
+      - name: Set up QEMU
+        uses: docker/setup-qemu-action@v3
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v3
+      - name: Login to Docker
+        uses: docker/login-action@v3
+        with:
+          registry: ghcr.io
+          username: ${{ github.actor }}
+          password: ${{ secrets.GITHUB_TOKEN }}
+      - name: Build and push
+        id: docker_build
+        uses: docker/build-push-action@v6
+        with:
+          context: ./infrastructure/images/terraform-azure-devops-agent
+          push: true
+          platforms: linux/amd64,linux/arm64
+          cache-from: gha
+          cache-to: gha
+          tags: ghcr.io/${{ steps.vars.outputs.reponame }}/terraform-azure-devops-agent:v${{ steps.vars.outputs.version }}
+        env:
+          SOURCE_DATE_EPOCH: ${{ env.TIMESTAMP }}
+      - name: Run Trivy vulnerability scanner
```

Comment:
This should happen before right? e.g. export to a tarball and run trivy on it. If it passes, then you push the image.
https://github.com/aquasecurity/trivy-action?tab=readme-ov-file#scanning-a-tarball
https://docs.docker.com/build/ci/github-actions/share-image-jobs/
https://docs.docker.com/reference/cli/docker/buildx/build/#output

Reply:
you are absolutely right! I will move the scanner job before push and also try to implement it in the pr pipeline.

Reply:
Should be implemented now, just need to ignore the kubectl alert so we can get an image built 😅

Reply:
Cloned kubectl repo on github and ran govulncheck to check if the reported CVE was an actual vulnerability. govulncheck didn't identify any calls to the affected code in golang.org/x/net so added it to the trivyignore file